src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -143,14 +143,21 @@ DatabaseShardImp::init()
                     j_)};
                 if (!shard->open(scheduler_, *ctx_))
                 {
-                    if (!shard->isLegacy())
-                        return false;
-
-                    // Remove legacy shard
                     shard->removeOnDestroy();
-                    JLOG(j_.warn()) <<
-                        "shard " << shardIndex <<
-                        " removed, legacy shard";
+                    if (!shard->isLegacy())
+                    {
+                        // Remove corrupted shard
```

Comment:
These comments (lines 149 and 156) are somewhat misleading as the removal isn't happening here.

Reply:
Done